### PR TITLE
MGMT-14633: Include manifest information in the log download

### PR DESCRIPTION
**Pull request by paul-maidment**
When the user downloads the logs for the cluster, we would like to include any cluster manifets in these logs so that they may be examined.

## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
  Manually set up a cluster with manifests and then triggered a download of the manifests, a CURL request similar to 
  `curl -X GET $SERVICE_URI/api/assisted-install/v2/clusters/61e94f28-bbd7-4719-b555-d7d7f9d5cc60/logs?logs_type=all --output logs.tar.gz` was used. 
  I also checked in the UI to ensure that the download looks appropriate.
- [] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/cluster/cluster.go [hunk 1]
```diff
@@ -1105,7 +1107,41 @@ func (m *Manager) createClusterDataFiles(ctx context.Context, c *common.Cluster,
...
 	}
+
+	includeSystemGenerated := true
+	listManifests, _ := m.manifestApi.ListClusterManifestsInternal(ctx, operations.V2ListClusterManifestsParams{ClusterID: *c.ID, IncludeSystemGenerated: &includeSystemGenerated})
+	for _, manifestEntry := range listManifests {
+		// Download the manifest content
+		path := filepath.Join(manifestEntry.Folder, manifestEntry.FileName)
+		objectName := filepath.Join(string(*c.ID), constants.ManifestFolder, path)
+		exists, _ := objectHandler.DoesObjectExist(ctx, objectName)
+		if exists {
```
[the remark this answers is not in the document]
Comment (paul-maidment) [reply internal/cluster/cluster.go:1118]: Good suggestion..taken!

[the remark this answers is not in the document]
Comment (carbonin) [reply internal/cluster/cluster.go:1118]: I think you want `!exists`, right?

**Review by carbonin — CHANGES_REQUESTED** [review]

File: internal/cluster/cluster.go [hunk 2]
```diff
@@ -1105,7 +1107,42 @@ func (m *Manager) createClusterDataFiles(ctx context.Context, c *common.Cluster,
...
+		manifest_prefix := "system-generated"
+		if err != nil {
+			log.WithError(err).Errorf("Unable to determine if manifest is user generated or not for %s/%s", manifestEntry.Folder, manifestEntry.FileName)
+			manifest_prefix = "unknown"
+		}
+		if isUserManifest {
+			manifest_prefix = "user-supplied"
+		}
+		fileName := fmt.Sprintf("%s/logs/manifest_%s_%s/%s", c.ID, manifest_prefix, manifestEntry.Folder, manifestEntry.FileName)
+		_ = m.uploadDataAsFile(ctx, log, string(fileContent), fileName, objectHandler)
```
Comment (carbonin) [inline internal/cluster/cluster.go:1142]: We should probably at least log here if this errors, right?

**Review by carbonin — APPROVED** [review]

**Pull request merged**